### DISCUSSION TITLE
Failed to reconstruct rve model from EBSD data

### DISCUSSION THREAD

**Discussion by Ming-is** [neperfepx/neper#307] [Q&A] [answered]
I try to introduce EBSD grain data into neper to establish an appropriate rve model.

![图片1](https://user-images.githubusercontent.com/71123582/138620544-92eadbd3-929e-4a60-85a7-2affee790cd2.png)

 The following are the data files I use, the corresponding instructions and error reports:
```
$ neper -T -n  from_morpho  -morpho  "size:file(sizefile),coo:file(coofile)" -dim 2 -ori "file(orifile,des=euler-bunge)" -oridescriptor "euler-bunge:active"   -domain "square(473 ,354)" -o simulation4

Info   :     [i] Parsing file `sizefile'...
Info   :     [i] Parsed file `sizefile'.
Abandoned (core dumped)
```

[neper.zip](https://github.com/rquey/neper/files/7406669/neper.zip)

I have EBSD ctf. file. How can I rebuild the rve？

**Comment by rquey** [comment, marked as answer]
The correct syntax is `-morpho  "size:file(sizefile),centroid:file(coofile)"` and not `-morpho  "size:file(sizefile),coo:file(coofile)"`.

You should then make sure that your centroids are consistent with the domain size. The centroid coordinates currently are in [0,57][0,43] while your domain is 473 x 354.

**Reply by Ming-is** [reply]
![image](https://user-images.githubusercontent.com/71123582/139078153-36eade62-3302-4191-a003-9d0e8b3394fb.png)
![image](https://user-images.githubusercontent.com/71123582/139079659-6a3da10a-7263-45e6-a14a-50277f9a392a.png)
This is the screenshot of the EBSD texture file I cut in ATEX software, so how do I set it in neper's "domain" instruction？

**Reply by rquey** [reply]
I would use micrometers and set the domain size using "-domain "square(18.84,14.16)". The grain centroids should be in micrometers too, and range in [0, 18.84][0, 14.16].

**Comment by Ming-is** [comment]
The orientation and coordinates of voxels can be obtained from the CTF file of EBSD. If I want to import it into the rve established by neper, what instructions can be used, and what appropriate instructions are available if I want to export the voxel information of rve？


Looking forward to your reply, thank you.

**Reply by rquey** [reply]
If you want to generate a Neper tessellation from grain properties (e.g . centroids), you have to write them in text files and use the commands discussed above.
If you want to describe your EBSD map as a Neper raster tessellation ("tesr") for further processing, you have to write it as a tesr file. See the documentation for details on the file format.

**Comment by Anmol-Bakshi** [comment]
@Ming-is what does size.txt represents here, like coofile.txt represents center of each grain co-ordinates; orifile contains orientation of each grain?
